src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2103,17 +2145,27 @@ public bool Delete()
                 bool handled = false;
                 if (!TextView.Selection.IsEmpty)
                 {
-                    if (TextView.Selection.Mode == TextSelectionMode.Stream || ReduceBoxSelectionToEditableBox())
+                    if (CutOrDeleteSelection(isCut: false))
                     {
-                        CutOrDeleteSelection(isCut: false);
                         MoveCaretToClosestEditableBuffer();
-                        handled = true;
                     }
+                    handled = true;
```

Comment:
@amcasey Not sure I understand your comment here. This method tries to decide if the given point lies with in the span of an active prompt, not a editable buffer (my understanding is all the prompts are in a single readonly buffer, which span over multiple spans).